Lib/unittest/mock.py

```diff
@@ -2401,3 +2420,35 @@ def __get__(self, obj, obj_type):
         return self()
     def __set__(self, obj, val):
         self(val)
+
+
+def seal(in_mock):
+    """Disables the automatic generation of "submocks"
+
+    Given an input Mock, seals it to ensure no further mocks will be generated
+    when accessing an attribute that was not already defined.
+
+    Submocks are defined as all mocks which were created DIRECTLY from the
+    parent. If a mock is assigned to an attribute of an existing mock,
+    it is not considered a submock.
+
+    :Example:
+
+    >>> mock = Mock()
+    >>> mock.submock.attribute1 = 2
+    >>> mock.not_submock = mock.Mock()
+    >>> seal(mock)
+    >>> mock.submock.attribute2  # This will raise
+    >>> mock.not_submock.attribute2  # This won't raise
+
+    """
+    in_mock._is_sealed = True
+    for attr in dir(in_mock):
+        try:
+            m = getattr(in_mock, attr)
+        except AttributeError:
+            continue
+        if not isinstance(m, NonCallableMock):
+            continue
+        if m._mock_new_parent is in_mock:
+            seal(m)
```

Comment:
Would it work to use _mock_methods and _mock_children attributes instead of dir()?